From 2efa09a318953a143ff8ef56d44945a68e3a8ce4 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Fri, 11 Aug 2023 21:49:02 -0500
Subject: [PATCH] Fix swap after updating util-linux

The most recent build of util-linux moved all of the binaries out of /bin and /sbin and into the corresponding /usr directory. Temporarily patch systemd to use the new paths.

This patch can be removed once T10615 is complete.
---
 src/core/swap.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/core/swap.c b/src/core/swap.c
index 2d25014e5f..5bcec64dcf 100644
--- a/src/core/swap.c
+++ b/src/core/swap.c
@@ -827,7 +827,7 @@ static void swap_enter_activating(Swap *s) {
                 }
         }
 
-        r = exec_command_set(s->control_command, "/sbin/swapon", "--fixpgsz", NULL);
+        r = exec_command_set(s->control_command, "/usr/sbin/swapon", "--fixpgsz", NULL);
         if (r < 0)
                 goto fail;
 
@@ -865,7 +865,7 @@ static void swap_enter_deactivating(Swap *s) {
         s->control_command = s->exec_command + SWAP_EXEC_DEACTIVATE;
 
         r = exec_command_set(s->control_command,
-                             "/sbin/swapoff",
+                             "/usr/sbin/swapoff",
                              s->what,
                              NULL);
         if (r < 0)
-- 
2.41.0

